dnf-behave-tests/common/lib/os_version.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import operator
+
+from behave.tag_matcher import ActiveTagMatcher
+
+def detect_os_version():
+    os_id = None
+    os_version = None
+
+    with open("/etc/os-release") as os_release:
+        for line in os_release:
+            if "=" not in line:
+                continue
+
+            k, v = line.split("=", 2)
+            if k == "ID":
+                os_id = v.strip(' \t\n"')
+            elif k == "VERSION_ID":
+                os_version = v.strip(' \t\n"')
+
+    # cut off any minor versions - in RHEL the version is e.g. 8.3, but we only want 8
+    os_version = os_version.split(".")[0]
+
+    # treat centos as RHEL in context of scenario tag matching
+    if os_id == "centos":
+        os_id = "rhel"
+
+    return os_id + "__" + os_version
```

Comment:
This could be simplified using `python3-distro` package (available on both fedora and rhel):
```
import distro
def detect_os_version():
    os_id = distro.id()
    if os_id == "centos": os_id = "rhel"
    os_version = distro.major_version()
    return os_id + "__" + os_version
```
```

Reply:
Much better, thanks :slightly_smiling_face: I did wonder if there's something, but thought it might not be on RHEL...